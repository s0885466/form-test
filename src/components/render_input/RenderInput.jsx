// @flow
import React from 'react';

type RenderInputProps = {
    type: string,
    name: string,
    label: string,
    placeholder: string,
    value: string,
    error: string,
    onChange: ()=>void,
}

function RenderInput({type, name, label, placeholder, value, onChange, error}: RenderInputProps) {
    return (
        <div>
            {label && <label htmlFor={name}>{label}</label>}
            <input type={type}
                   onChange={onChange}
                   id={name}
                   name={name}
                   value={value}
                   placeholder={placeholder}
                   className={error && 'error'}
            />
            {error && <div className={'errors'}>{error}</div>}
        </div>
    )
}

export default RenderInput;