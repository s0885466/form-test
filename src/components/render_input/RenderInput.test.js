import React from 'react';
import {shallow, mount, render} from 'enzyme';
import RenderInput from "./RenderInput";

describe('Testing snapshop <RenderInput/>', () => {
    it('App have rendered correctly', () => {
        const renderInput = shallow(<RenderInput/>);
        expect(renderInput).toMatchSnapshot();
    })
});

describe('Handlers tests <RenderInput/>', () => {
    it('onChange input name invokes a function', () => {
        const change = jest.fn();
        const renderInput = shallow(<RenderInput
            name={'name'}
            label={'Name'}
            placeholder={'type name'}
            value={''}
            onChange={change}
            error={''}/>);
        renderInput.find('#name').simulate('change', {target: {value: "spam"}});
        expect(change).toHaveBeenCalledWith({target: {value: "spam"}});
    })
});

describe('Show error or doesnt show error', () => {

    it('error exists' , () => {
        let error = 'ERROR';
        const renderInput = mount(<RenderInput
            name={'name'}
            label={'Name'}
            placeholder={'type name'}
            value={''}
            onChange={()=>{}}
            error={error}/>);
        expect(renderInput.exists('input.error')).toBeTrue();
        expect(renderInput.find('div.errors').text()).toBe(error);
    });

    it('error doesnt exist' , () => {
        let error = '';
        const renderInput = mount(<RenderInput
            name={'name'}
            label={'Name'}
            placeholder={'type name'}
            value={''}
            onChange={()=>{}}
            error={error}/>);
        expect(renderInput.exists('input.error')).toBeFalse();
        expect(renderInput.exists('div.errors')).toBeFalse();
    });
});
