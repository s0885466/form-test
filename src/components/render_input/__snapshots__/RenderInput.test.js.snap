// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshop <RenderInput/> App have rendered correctly 1`] = `ShallowWrapper {}`;
